### DISCUSSION TITLE
Help request for failures GitHub pages

### DISCUSSION THREAD

**Discussion by FrancoisPapin** [community/community#31073] [Other Feature Feedback, Questions, & Ideas]
Hi,

I am not a specialist in coding. I have some failures with Github-pages for my site : 
--
https://github.com/FrancoisPapin/NPEi.lab
--

So I couldn't yet change anything or actualize the site.

I implore the community to help me. Please, could you help me?

Thanks

François

**Comment by airtower-luna** [comment]
If you look at the run summary of your site getting built, you can see [the run crashed with a "No space left on device" error](https://github.com/FrancoisPapin/NPEi.lab/actions/runs/2940603709). And [looking at the checkout step log](https://github.com/FrancoisPapin/NPEi.lab/runs/8052961357?check_suite_focus=true#step:3:648), well:

> `Receiving objects: 100% (7702/7702), 7.14 GiB | 29.67 MiB/s, done.`

And that's with only fetching the latest commit. Add to that the checked out files, and Jekyll copying files while building the site. According to the documentation the GitHub hosted runners [provide 14 GB of disk space](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources).

Conclusion: Your site is *way* too large for the build infrastructure to handle. :sweat_smile:

But even if the infrastructure could handle it, it's also significantly bigger than the [limit for Pages sites](https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#usage-limits):

> Published GitHub Pages sites may be no larger than 1 GB.

So I'm afraid you'll either have to significantly shrink your site, or find another hoster.

**Reply by FrancoisPapin** [reply]
Thank you for your answer.
So, if I suppress much data for more memory, the site will run again. Do you think that?

**Reply by airtower-luna** [reply]
Assuming there isn't any other blocking problem, yes. :slightly_smiling_face:

Mind that if you just delete files they're still part of the repository history, so if you clean up you might want to clear them out of the repository history too, to make the repository easier to handle. Git can handle larger repositories, too, but any clone etc. will need to get the whole (large) repository history. And for repositories on GitHub keeping them under 5GB total [is "strongly recommended"](https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github#repository-size-limits). That documentation also explains how you can clean the history. Make sure to have a backup of the files you remove though, assuming you want to keep them for yourself, if not on the site. :wink:

**Comment by radar-blip** [comment]
This is your site? https://francoispapin.github.io/NPEi.lab/

**Reply by FrancoisPapin** [reply]
Oh yes.  In fact, a website project.

**Reply by radar-blip** [reply]
Ok, I guess I am confused here. 
Anyway, just listen to airtower-luna, she knows everything there is to know. 👍🏻
